### PR TITLE
feat(calendar,date-picker): add month-year picker and all date inputs

### PR DESCRIPTION
**Pull request by MerlinMoos**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our
      guidelines: https://github.com/spartan-ng/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### Primitives

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] autocomplete
- [ ] avatar
- [ ] badge
- [ ] breadcrumb
- [ ] button
- [ ] button-group
- [x] calendar
- [ ] card
- [ ] carousel
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [x] date-picker
- [ ] dialog
- [ ] empty
- [ ] dropdown-menu
- [ ] field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] input-group
- [ ] input-otp
- [ ] item
- [ ] kbd
- [ ] label
- [ ] menubar
- [ ] native-select
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] resizable
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] sidebar
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toggle
- [ ] toggle-group
- [ ] tooltip
- [ ] typography

### Others

- [ ] trpc
- [ ] nx
- [ ] repo
- [ ] cli
- [ ] mcp

## What is the current behavior?

Closes #

## What is the new behavior?

This pull request introduces new calendar and date picker examples, enhances the demo pages to showcase these new features, and improves the date input formatting for better user experience. It also adds comprehensive end-to-end (E2E) tests to ensure robust coverage of date picker interactions and edge cases.

**New Examples and Demo Enhancements:**

* Added a `CalendarYearAndMonthDropdownsExample` component to demonstrate calendar caption layouts with dropdowns for month and year selection, and updated the calendar demo page to include and display this example separately from the basic year/month calendar (`calendar--year-and-month-dropdowns.example.ts`, `calendar.page.ts`). [[1]](diffhunk://#diff-30d239dbea679fcf47b8ca4416e908a1e9ce66acbd5788197964d19548008c91R1-R45) [[2]](diffhunk://#diff-73813a52f49cd2a8fb1266e117647ca052bae478ddc7d5088b80013cb4a88d7eR6) [[3]](diffhunk://#diff-73813a52f49cd2a8fb1266e117647ca052bae478ddc7d5088b80013cb4a88d7eL55-R58) [[4]](diffhunk://#diff-73813a52f49cd2a8fb1266e117647ca052bae478ddc7d5088b80013cb4a88d7eL154-R165) [[5]](diffhunk://#diff-73813a52f49cd2a8fb1266e117647ca052bae478ddc7d5088b80013cb4a88d7eR208)
* Introduced a `DatePickerYearMonthExample` component and integrated it into the date picker demo page to demonstrate a picker for selecting only month and year (`date-picker--month-year.example.ts`, `date-picker.page.ts`). [[1]](diffhunk://#diff-3b41e1b710d385e4dd6bf55cb434815e6e0d8f4940fc09e5ddd29ec56a1f21cbR1-R24) [[2]](diffhunk://#diff-0276a9e5d387698537a24a29611603b54f39414a3ebc0fddcd8f1ec2f9f981e5R7) [[3]](diffhunk://#diff-0276a9e5d387698537a24a29611603b54f39414a3ebc0fddcd8f1ec2f9f981e5R76) [[4]](diffhunk://#diff-0276a9e5d387698537a24a29611603b54f39414a3ebc0fddcd8f1ec2f9f981e5R218-R225) [[5]](diffhunk://#diff-0276a9e5d387698537a24a29611603b54f39414a3ebc0fddcd8f1ec2f9f981e5R322)

**Date Input Formatting Improvements:**

* Updated the date picker input example so that the display format shown after committing a date is more user-friendly (e.g., 'Sat Jun 15 2024'), while the input format for editing remains 'dd.MM.yyyy'. This improves clarity for users when entering and viewing dates (`date-picker--input.example.ts`). [[1]](diffhunk://#diff-9328caa110ccb0cca27e2486499b8662fb3d789536926ddfbf47be225471cdbdL18-R23) [[2]](diffhunk://#diff-9328caa110ccb0cca27e2486499b8662fb3d789536926ddfbf47be225471cdbdL31-R37)

**Testing and Coverage:**

* Added a comprehensive E2E test suite for the date picker family, covering single, multi, range, and year-month pickers, as well as all input variants. The tests include checks for caret position, commit-on-blur, Enter/blur regression, invalid input handling, and accessibility (`date-picker.cy.ts`).

**Refactoring and Code Organization:**

* Refactored the existing year/month calendar example to simplify its implementation, improve naming, and add i18n documentation and usage examples (`calendar--year-and-month.example.ts`).

These changes collectively improve the documentation, usability, and reliability of the calendar and date picker components.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Summary by CodeRabbit

* **New Features**
  * Added month/year picker examples and Storybook demos for date, range, multi-date, and month/year input flows.
  * Introduced month/year calendar and picker UI support, including navigation, selection, and configurable input formatting.
  * Added new date entry variants for single, range, multi-date, and month/year selection.
* **Bug Fixes**
  * Improved input parsing and display formatting so typed dates stay consistent with what users see.
  * Preserved user-entered text during invalid input and improved commit behavior on blur/Enter.

### CODE REVIEW

**Comment by MerlinMoos** [comment]
@greptile and @coderabbit review again

**Comment by goetzrobin** [comment]
### Optional: dedup the shared TS across the four date inputs

Really nice work on the `BrnDateInput` headless base — that's the right shape. One follow-on idea, totally your call.

The four styled inputs (`hlm-date-picker-input`, `hlm-date-multi-input`, `hlm-date-range-input`, `hlm-month-year-input`) share an identical presentational surface: `placeholder`, `forceInvalid`, `clearAriaLabel`, `calendarAriaLabel`, the `inputId` + `_nextId` counter, and `triggerId`. That block is copy-pasted in all four (and again in the CLI templates).

Could hoist just that into a thin helm-level base between `BrnDateInput` and the concretes:

```ts
@Directive()
export abstract class HlmDateInputBase<V> extends BrnDateInput<V> implements BrnDatePickerTriggerBase {
	private static _nextId = 0;

	public readonly inputId = input(`hlm-date-input-${HlmDateInputBase._nextId++}`);
	public readonly placeholder = input('');
	public readonly forceInvalid = input<boolean, BooleanInput>(false, { transform: booleanAttribute });
	public readonly clearAriaLabel = input<string>('Clear date');
	public readonly calendarAriaLabel = input<string>('Open calendar');

	/** @internal Id used by the trigger contract for labeling. */
	public readonly triggerId = this.inputId;
}
```

Each concrete then keeps its own inline `@Component` template (see caveat below) and shrinks to just: config injection + `parseDate`/`formatInputDate(s)` inputs + the two `parseValue`/`formatInputValue` overrides.

**Deliberate non-goal:** I'm *not* suggesting extracting the shared `template` into a string const — that would kill template type-checking and the language service, so the identical inline templates should stay as-is. That means this is a modest win (the TS boilerplate, not the markup), which is why it's fully optional. If you'd rather keep all four self-contained for readability, that's a legitimate call too — trusting your judgement here.

File: tsconfig.base.json [hunk 1]
```diff
@@ -137,7 +137,8 @@
 			"@spartan-ng/mcp": ["libs/mcp/src/index.ts"],
 			"@spartan-ng/registry": ["libs/registry/src/index.ts"],
 			"@spartan-ng/tools": ["libs/tools/src/index.ts"],
-			"@spartan-ng/trpc": ["libs/trpc/src/index.ts"]
+			"@spartan-ng/trpc": ["libs/trpc/src/index.ts"],
+			"@spartan-ng/brain/date-picker": ["./libs/brain/date-picker/src/index.ts"]
```
Comment (goetzrobin) [inline tsconfig.base.json]: Nit: `./` prefix — every other path in this map is root-relative without it (`libs/...`).

```suggestion
			"@spartan-ng/brain/date-picker": ["libs/brain/date-picker/src/index.ts"]
```

File: libs/brain/tsconfig.lib.json [hunk 2]
```diff
@@ -8,6 +8,15 @@
...
+		"**/*.spec.ts",
+		"test-setup.ts",
+		"vite.config.mts",
+		"**/*.test.ts",
+		"src/testing/**",
+		"date-picker/**/*.spec.ts",
+		"date-picker/**/*.test.ts",
+		"date-picker/src/testing/**"
+	],
+	"include": ["**/*.ts", "date-picker/**/*.ts"]
```
Comment (goetzrobin) [inline libs/brain/tsconfig.lib.json]: `**/*.ts` already matches `date-picker/**/*.ts`, and `**/*.spec.ts` / `**/*.test.ts` / `src/testing/**` already cover the three `date-picker/**` excludes above. Are these extra globs load-bearing, or leftover? If date-picker resolves without them, drop all four.

File: libs/brain/date-picker/src/lib/brn-date-input.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+		if (!value) {
+			this._datePicker.updateDate?.(undefined);
+			this._datePicker.touched?.();
+			return;
+		}
+
+		// Invalid parse: clear the picker value, keep the text so the user can fix it.
+		const parsed = this.parseValue(value);
+		this._datePicker.updateDate?.(parsed ?? undefined);
```
Comment (goetzrobin) [inline libs/brain/date-picker/src/lib/brn-date-input.ts]: `parseValue` returns `V | undefined` (never `null`), so `?? undefined` is a no-op.

```suggestion
		this._datePicker.updateDate?.(parsed);
```

File: libs/helm/date-picker/src/lib/hlm-month-year-picker.token.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			return `${month}/${year}`;
+		},
+		formatInputDate: (date) => {
+			if (!(date instanceof Date)) return `${date}`;
+
+			const month = String(date.getMonth() + 1).padStart(2, '0');
+			const year = date.getFullYear();
+
+			return `${month}/${year}`;
+		},
```
Comment (goetzrobin) [inline libs/helm/date-picker/src/lib/hlm-month-year-picker.token.ts]: `formatInputDate` is byte-identical to `formatDate` above. If month-year's display and edit formats are intentionally the same, pull the body into one local (`const mmYYYY = (date) => ...`) and reference it from both instead of duplicating.

File: libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts [hunk 5]
```diff
@@ -14,20 +14,77 @@ export interface HlmDatePickerMultiConfig<T> {
...
+	 * @param value the raw string from the input
+	 * @returns the parsed date, or `undefined` when the value can't be parsed
+	 */
+	parseDate: (value: string) => T[] | undefined;
 }
 
 function getDefaultConfig<T>(): HlmDatePickerMultiConfig<T> {
 	return {
 		formatDates: (dates) => dates.map((date) => (date instanceof Date ? date.toDateString() : `${date}`)).join(', '),
+		formatInputDates: (dates) =>
```
Comment (goetzrobin) [inline libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts:46]: Heads up on cross-variant consistency: multi's default edit format is numeric `dd/MM/yyyy`, but single (`hlm-date-picker.token.ts` L46) and range both default to `toDateString()` (`Sat Jun 15 2024`). Intentional? Three input variants shipping three different edit formats by default is surprising.

File: libs/helm/date-picker/src/lib/hlm-date-range-picker.token.ts [hunk 6]
```diff
@@ -30,8 +47,26 @@ function getDefaultConfig<T>(): HlmDateRangePickerConfig<T> {
...
+		parseDate: (value) => {
+			if (typeof value !== 'string') return undefined;
+
+			const parts = value.split(' - ').map((part) => part.trim());
+			if (parts.length !== 2) return undefined;
+
+			const start = new Date(parts[0]);
+			const end = new Date(parts[1]);
+
+			if (isNaN(start.getTime()) || isNaN(end.getTime())) return undefined;
```
Comment (goetzrobin) [inline libs/helm/date-picker/src/lib/hlm-date-range-picker.token.ts]: Round-trip concern: `formatInputDates` does `.filter(Boolean)`, so a `[start, undefined]` model formats to a single token with no ` - ` separator. This `parseDate` then splits on ` - `, gets length 1, and returns `undefined` → the value is cleared on the next focus→blur. If the range model can ever hold a partial range, that's a silent data-loss round-trip. Worth a test.

**Review by goetzrobin — COMMENTED** [review]
A few surgical follow-ups from a close read — mostly nits + two questions, nothing blocking.

File: libs/brain/tsconfig.lib.json (hunk 2 above)
Comment (marcjulian) [reply libs/brain/tsconfig.lib.json]: This is automatically added when you use `nx g @spartan-ng/tools:brain-secondary-entrypoint --name=...` or the helm command. @ashley-hunter [fixed](https://github.com/spartan-ng/spartan/pull/1498) that for the spartan CLI, as that happened there as well. Perhaps we can disable that here as well.

File: libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts (hunk 5 above)
Comment (MerlinMoos) [reply libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts:46]: yes that was intentional, I wanted to keep the old format for this. We can align it but at the end, people can change the format anyways. We can align a format if you want?

File: libs/helm/date-picker/src/lib/hlm-date-range-picker.token.ts (hunk 6 above)
Comment (MerlinMoos) [reply libs/helm/date-picker/src/lib/hlm-date-range-picker.token.ts]: great catch, I will set the startDate <=> endDate in case on of them is undefined

File: libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts (hunk 5 above)
Comment (goetzrobin) [reply libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts:46]: I guess aligning would be breaking changes? or no? I guess its just default so not super important

**Comment by goetzrobin** [comment]
lots of failures for some reason @MerlinMoos

File: libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts (hunk 5 above)
Comment (MerlinMoos) [reply libs/helm/date-picker/src/lib/hlm-date-picker-multi.token.ts:46]: can be a breaking change when we change that.
